src/app/user-settings/settings/page.tsx

```diff
@@ -1,3 +1,269 @@
+          <h3 className="text-xl font-semibold">INTERESTS</h3>
+          <InterestIcon className="h-7 w-7" />
+        </div>
+        <p className="mb-4 text-lg text-stone-500">
+          Pick the curated content you want to see most.
+        </p>
+        <div className="grid max-h-64 auto-rows-max grid-cols-1 gap-2 overflow-auto rounded-md border bg-white p-4 md:grid-cols-2">
+          <div className="flex items-center gap-1">
+            <input type="checkbox" value="Chemistry" className="mr-2" />
+            <ChemistryIcon className="w-5" />
+            Chemistry
+          </div>
+          <div className="flex items-center gap-1">
+            <input type="checkbox" value="Chemistry" className="mr-2" />
+            <CeIcon className="w-6" />
+            Chemical Eng.
+          </div>
+          <div className="flex items-center gap-1">
+            <input type="checkbox" value="Chemistry" className="mr-2" />
+            <MeIcon className="w-6" />
+            Mechanical Eng.
+          </div>
+          <div className="flex items-center gap-1">
+            <input type="checkbox" value="Chemistry" className="mr-2" />
+            <MathIcon className="w-6" />
+            Mathematics
+          </div>
+          <div className="flex items-center gap-1">
+            <input type="checkbox" value="Chemistry" className="mr-2" />
+            <PhysicsIcon className="w-6" />
+            Physics
+          </div>
+          <div className="flex items-center gap-1">
+            <input type="checkbox" value="Chemistry" className="mr-2" />
+            <SociologyIcon className="w-6" />
+            Sociology
+          </div>
+          <div className="flex items-center gap-1">
+            <input type="checkbox" value="Chemistry" className="mr-2" />
+            <CsIcon className="w-6" />
+            Comp Sci.
+          </div>
+          <div className="flex items-center gap-1">
+            <input type="checkbox" value="Chemistry" className="mr-2" />
+            <EeIcon className="w-6" />
+            Electrical Eng.
+          </div>
+          <div className="flex items-center gap-1">
+            <input type="checkbox" value="Chemistry" className="mr-2" />
+            <EnvSciIcon className="w-6" />
+            Envir. Science
+          </div>
+          <div className="flex items-center gap-1">
+            <input type="checkbox" value="Chemistry" className="mr-2" />
+            <MedicineIcon className="w-6" />
+            Medicine
+          </div>
+          <div className="flex items-center gap-1">
+            <input type="checkbox" value="Chemistry" className="mr-2" />
+            <PsyIcon className="w-6" />
+            Psychology
+          </div>
+          <div className="flex items-center gap-1">
+            <input type="checkbox" value="Chemistry" className="mr-2" />
+            <TechIcon className="w-6" />
+            Technology
+          </div>
+        </div>
+      </section>
```

Comment:
I believe that we can generate these from the prisma enum rather than hard code each of the rows

Reply:
Are you suggesting using something like an array/object of topics (like StoryTopic in prisma?) and .map() method to generate the list? I think it will need extra code to map each topic to its corresponding icon, so for now I just hard code the topics with their icon here. Do you have any suggestions on mapping icons from topics?

Reply:
One suggestion I have is to isolate the icons into a separate component which allows you to map the topic to the correct icon. You would write something like this:

```typescript
const iconMap = {
  "ASTRONOMY": <AstronomyIcon />,
...
}

return iconMap[topic];
```

This is so that we can reduce hard-coded sections. In the event that we add additional topics, we would simply augment the iconMap object with the new icon component.

Reply:
I made an edit to the code block above that now includes the `Record` type in the object definition. This has an advantage: it requires every StoryTopic item to have an entry in the object. If for some reason, we add a new topic but forget to update this object, TS will throw a type error.

Reply:
Sounds great. I will make a change based on the above idea.

Reply:
All done.